zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanTask.java

```diff
@@ -119,13 +119,11 @@ public void run() {
                                             .contains(hrefHistoryType))) {
 
                         if (scanner instanceof PluginPassiveScanner) {
-                            PluginPassiveScanner pps = (PluginPassiveScanner) scanner;
-                            PluginPassiveScanner pps2 = pps.copy();
-                            pps2.init(psThread, msg, passiveScanData);
-                            scanner = pps2;
-                        } else {
-                            scanner.setParent(psThread);
+                            PluginPassiveScanner pps = ((PluginPassiveScanner) scanner).copy();
+                            pps.setHelper(passiveScanData);
+                            scanner = pps;
                         }
+                        scanner.setParent(psThread);
```

Comment:
Do we always want to do this? It wasn't being set before if scanner was not an instance of `PluginPassiveScanner`.

Reply:
Disregard, just looked at `init`.